public/type_templates/glb.js

```diff
@@ -44,16 +44,16 @@ export default ctx => {
     useLoaders,
     usePhysics,
     usePhysicsTracker,
-    useLocalPlayer
+    useLocalPlayer,
+    useRenderer
   } = ctx;
 
   const app = useApp();
   
   const {gltfLoader, exrLoader} = useLoaders();
   // const renderer = useRenderer();
   const engine = useEngine();
-  const {webaverseRenderer} = engine;
-  const {renderer} = webaverseRenderer;
+  const renderer = engine ? engine.webaverseRenderer.renderer : useRenderer(); // companion app does not have engine

```

Comment:
Why not always do `useRenderer()`? Why sometimes peel from the engine?

Reply:
[only use useRenderer](https://github.com/upstreet-labs/compiler/pull/3/commits/3638f9c70db6dc3b4e91b16d5b61ae9cefe33700)